File: src/Plotly.NET/Traces/ObjectAbstractions/Gradient.fs
namespace Plotly.NET.TraceObjects

open Plotly.NET
open Plotly.NET.LayoutObjects
open DynamicObj
open System
open System.Runtime.InteropServices

type Gradient() =
    inherit DynamicObj()

    static member init
        (
            ?Color: Color,
            ?Type: StyleParam.GradientType,
            ?MultiTypes: seq<StyleParam.GradientType>
        ) =
        Gradient() |> Gradient.style (?Color = Color, ?Type = Type, ?MultiTypes = MultiTypes)

    static member style
        (
            ?Color: Color,
            ?Type: StyleParam.GradientType,
            ?MultiTypes: seq<StyleParam.GradientType>
        ) =

        fun (gradient: Gradient) ->

            gradient
            |> DynObj.withOptionalSingleOrMultiProperty "type" (Type, MultiTypes)
            |> DynObj.setOptionalProperty "color" Color

File: src/Plotly.NET.CSharp/ObjectAbstractions/TraceObjects/Gradient.cs

using Microsoft.FSharp.Core;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Plotly.NET.CSharp;

public static partial class TraceObjects
{
    public static Plotly.NET.TraceObjects.Gradient Gradient
        (

            Optional<Color> Color = default,
            Optional<StyleParam.GradientType> Type = default,
            Optional<IEnumerable<StyleParam.GradientType>> MultiTypes = default
        ) => 
            Plotly.NET.TraceObjects.Gradient.init(

                Color: Color.ToOption(),
                Type: Type.ToOption(),
                MultiTypes: MultiTypes.ToOption()
            );
}

